app/screens/settings-screen/username-screen.tsx

```diff
@@ -202,18 +186,24 @@ export const UsernameScreen = ({ navigation }: Props) => {
         leftIcon={{ type: "ionicon", name: "ios-person-circle" }}
         onChangeText={onChangeText}
         errorStyle={styles(messageIsError).error}
-        errorMessage={message}
+        errorMessage={(() => {
+          // Wait for everything to download and sync through useEffect before displaying
+          if (
+            !loadingUserNameExist &&
+            (messageIsError || messageIsError === usernameExists)
```

Comment:
This boolean expression is hard to parse. How do we ever get to `messageIsError === usernameExists`? If `messageIsError` is defined the first part of the `||` will already be `true` right?

Reply:
The problem is messageIsError is updated by a useEffect watching data (data is what gets parsed into usernameExists). So there was a flicker between react renderings where usernameExists could be true and messageIsError could be false or vice versa.

I could give this boolean expression a name or we could reduce the use of useEffect.